### PR TITLE
go: add context page

### PR DESCRIPTION
**Pull request by kopiczko**
Fixes https://github.com/giantswarm/giantswarm/issues/4749.

### CODE REVIEW

File: go/context.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Context
+
+We use `context.Context` ([godoc][context]) as a first argument in our APIs.
+This allows us to:
+
+- Use `micrologger.Logger.LogCtx`.
+- Make sure we pass the `context.Context` down the stack if we happen to
+  distribute information in it.
+- Allows use to handle cancellation and deadlines without changing the API.
```
Comment (xh3b4sd) [inline go/context.md]:
```suggestion
- Allows us to handle cancellation and deadlines without changing the API.
```

File: go/context.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+Guidelines:
+
+- Don't pass `context.Context` to `New*` functions.
+- Don't pass `context.Context` to getter and setter functions.
+- Don't use `context.TODO()` in tests. It is not TODO. We are not going to
+  change that.
+- Otherwise always make `ctx context.Context` a first argument to the exported
+  function of method.
```
Comment (xh3b4sd) [inline go/context.md]:
```suggestion
  function or method.
```

**Review by xh3b4sd — APPROVED** [review]

**Review by tuommaki — APPROVED** [review]
I'm not personally that fond of context everywhere, but I can definitely see the benefit for logging so let's see how this go. :+1:

**Review by indyfree — APPROVED** [review]
Nice, good to have some Guidelines with it aswell!

**Review by MarcelMue — APPROVED** [review]

**Pull request merged**